db/utils.js

```diff
@@ -49,6 +49,24 @@ const DBUtils = {
     }
   },
 
+  async _getSha1EntriesFromPrefixAndDo(sha1Prefix, aFoundCallback, aNotFoundCallback) {
+    // Only accept 6-character hash prefixes so requests:
+    // 1. can't get more than the intended hash range results FROM us
+    // 2. can't reveal a more specific hash query TO us
+    if (sha1Prefix.length !== 6) {
```

Comment:
Now that `6` is defined in 2 places (in 2 different files), not sure if we should give it a `const` in AppConstants, then tweak this to something like:

```js
if (sha1Prefix.length !== AppConstants.SHA_PREFIX_LENGTH) {
  // ...
}
```

Reply:
It might also be worth considering putting all the HIBP stuff in one file (a utils file?), and require that wherever it's necessary. No strong feelings about this though.

Edit: never mind, things got a bit clearer as I read more of the code in this PR